src/main/java/gregtech/api/fluids/definition/FluidDefinition.java

```diff
@@ -0,0 +1,333 @@
+    }
+
+    protected int getInferredTemperature() {
+        if (this.state == FluidState.PLASMA) {
+            return FluidConstants.PLASMA_TEMPERATURE;
+        }
+        return FluidConstants.AMBIENT_TEMPERATURE;
+    }
+
+    /**
+     * @param fluidName the name of the fluid to create
+     * @return the new fluid
+     */
+    @Nonnull
+    public Fluid constructFluid(@Nonnull String fluidName) {
+        if (this.color == -1) this.color = 0xFFFFFF;
+        return new AdvancedFluid(fluidName, color, this);
+    }
+
+    @Nonnull
+    @Override
+    public FluidState getState() {
+        return state;
+    }
+
+    @Nonnull
+    @Override
+    public Collection<FluidTag> getTags() {
+        return tags;
+    }
+
+    /**
+     * @param tag the tags to add
+     */
+    public void addTag(@Nonnull FluidTag tag) {
+        this.tags.add(tag);
+    }
+
+    /**
+     * @param tags the tags to add
+     */
+    public void addTags(@Nonnull FluidTag... tags) {
+        this.tags.addAll(Arrays.asList(tags));
+    }
+
+    @Override
+    public int getTemperature() {
+        return temperature;
+    }
+
+    /**
+     * @param temperature the temperature to set
+     */
+    public void setTemperature(int temperature) {
+        Preconditions.checkArgument(temperature > 0 || temperature == -1,
+                "Temperature must be > 0, or -1 for inferred values.");
+        if (temperature == -1) this.temperature = getInferredTemperature();
+        else this.temperature = temperature;
+    }
+
+    /**
+     * @return the translation key for this fluid
+     */
+    @Nonnull
+    public String getTranslationKey() {
+        return translationKey;
+    }
+
+    /**
+     * @param translationKey the translation key to set
+     */
+    public void setTranslationKey(@Nonnull String translationKey) {
+        this.translationKey = translationKey;
+    }
+
+    /**
+     * @return the still texture
+     */
+    @Nonnull
+    public ResourceLocation getStill() {
+        return still;
+    }
+
+    /**
+     * @param still the texture to set
+     */
+    public void setStill(@Nonnull ResourceLocation still) {
+        this.still = still;
+    }
+
+    /**
+     * @return the flowing texture
+     */
+    @Nonnull
+    public ResourceLocation getFlowing() {
+        return flowing;
+    }
+
+    /**
+     * @param flowing the texture to set
+     */
+    public void setFlowing(@Nonnull ResourceLocation flowing) {
+        this.flowing = flowing;
+    }
+
+    /**
+     * @return the color of the fulid
+     */
+    public int getColor() {
+        return color;
+    }
+
+    /**
+     * @param color the color to set
+     */
+    public void setColor(int color) {
+        this.color = color;
+    }
+
+    /**
+     *
+     * @return if this fluid has a block
+     */
+    public boolean hasBlock() {
+        return hasBlock;
+    }
+
+    /**
+     * @param hasBlock the state to set
+     */
+    public void setHasBlock(boolean hasBlock) {
```

Comment:
Why do we have setters for these if they are already set in the builder?

Reply:
Setters are needed for mods modifying the fluids of other mods. The builders for completed FluidDefinitions are not easily accessible from other mods.

Reply:
Modify other mods fluids, or modify fluids in an addon mod?